samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java

```diff
@@ -207,7 +207,11 @@ static public Object convertToAvroObject(Object relObj, Schema schema) {
             .collect(Collectors.toMap(Map.Entry::getKey, e -> convertToAvroObject(e.getValue(),
                 getNonNullUnionSchema(schema).getValueType())));
       case UNION:
-        return convertToAvroObject(relObj, getNonNullUnionSchema(schema));
+        for (Schema unionSchema : schema.getTypes()) {
```

Comment:
We have an AvroRelConverter version in Linkedin. We will have to merge these two and get rid of the Linkedin one.

Reply:
I already updated the linkedIn version to unblock the customer. I will open a bug to remove the LI one and use the open source version.